x-pack/plugins/observability_solution/observability_onboarding/server/routes/types.ts

```diff
@@ -50,3 +51,34 @@ export interface ObservabilityOnboardingRouteCreateOptions {
     xsrfRequired?: boolean;
   };
 }
+
+export const IntegrationRT = t.union([
+  t.type({
+    pkgName: t.string,
+    installSource: t.literal('registry'),
+  }),
+  t.type({
+    pkgName: t.string,
+    installSource: t.literal('custom'),
+    logFilePaths: t.array(t.string),
+  }),
+]);
```

Comment:
Instead of first creating the schema for your saved object using `@kbn/config-schema` and then creating the corresponding typescript interface using `io-ts` to match them you can infer the types using `TypeOf`:

```
import type { TypeOf } from '@kbn/config-schema';

export type ObservabilityOnboardingFlow = TypeOf<typeof savedObjectSchema>;
```

`savedObjectSchema` here is any schema created using `@kbn/config-schema`

Reply:
I needed `io-ts` types for [route body validation here](https://github.com/elastic/kibana/pull/184717/files#diff-03446290b1892d7b9befa59354373be0f2c04035c683a2b97d977c22aee0a98dR73), not really to match the schema. It's a bit annoying to have two runtime types in this case, it's unfortunate there is no easy way to convert from one to another.

Reply:
Ahh, got you! Yeh, one of the reasons I'm not a fan of these bespoke abstractions in Kibana. We're doing a lot of work here just to avoid using the platform. 🤷